pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go

```diff
@@ -0,0 +1,645 @@
+								{
+									Name:      DeploymentName,
+									MountPath: MountPathVpnSeedServer,
+								},
+								{
+									Name:      VpnSeedServerTLSAuth,
+									MountPath: MountPathTLSAuth,
+								},
+								{
+									Name:      VpnSeedServerDH,
+									MountPath: MountPathDH,
+								},
+							},
+						},
+						{
+							Name:            "envoy-proxy",
+							Image:           v.imageAPIServerProxy,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							SecurityContext: &corev1.SecurityContext{
+								Capabilities: &corev1.Capabilities{
+									Add: []corev1.Capability{
+										"NET_BIND_SERVICE",
+									},
+								},
+							},
+							Command: []string{
+								"envoy",
+								"--concurrency",
+								"2",
+								"-c",
+								"/etc/envoy/envoy.yaml",
+							},
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse("20m"),
+									corev1.ResourceMemory: resource.MustParse("20Mi"),
+								},
+								Limits: corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse("200m"),
+									corev1.ResourceMemory: resource.MustParse("300Mi"),
+								},
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "envoy-config",
+									MountPath: "/etc/envoy",
+								},
+								{
+									Name:      DeploymentName,
+									MountPath: "/etc/tls",
+								},
+							},
+						},
+					},
+					TerminationGracePeriodSeconds: pointer.Int64Ptr(30),
+					Volumes: []corev1.Volume{
+						{
+							Name: DeploymentName,
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: DeploymentName,
+								},
+							},
+						},
+						{
+							Name: VpnSeedServerTLSAuth,
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: VpnSeedServerTLSAuth,
+								},
+							},
+						},
+						{
+							Name: VpnSeedServerDH,
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: VpnSeedServerDH,
+								},
+							},
+						},
+						{
+							Name: "envoy-config",
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: EnvoyConfig,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		if v.nodeNetwork != nil {
+			deployment.Spec.Template.Spec.Containers[0].Env = append(
+				deployment.Spec.Template.Spec.Containers[0].Env,
+				corev1.EnvVar{Name: "NODE_NETWORK", Value: *v.nodeNetwork},
+			)
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, v.client, configMap, func() error {
+		configMap.Labels = map[string]string{
+			v1beta1constants.GardenRole:     v1beta1constants.GardenRoleSystemComponent,
```

Comment:
Why do you set this label? All of this is deployed into the seed, so it's part of the control plane and not a system component?

Reply:
Not sure why we added it. It will be removed.